apps/system/js/browser_context_menu.js

```diff
@@ -171,6 +193,31 @@
       }));
     },
 
+    pinUrl: function(url) {
+      var data = {
+        type: 'url',
+        url: url,
+        iconable: false,
+        title: this.app.title,
+        theme: this.app.themeColor
+      };
+
+      this.app.getScreenshot(function() {
+        data.screenshot = this.app._screenshotBlob;
+        this.app.getSiteIconUrl()
```

Comment:
Strictly speaking this may not be the icon which actually gets displayed on the homescreen because the homescreen only gets icons from the places database which doesn't include manifest icons. We're hoping to eventually fix this by adding support for manifest icons to pinned pages but until then not sure if it's better to make this consistent or leave it inconsistent but how we want it to be later?